packages/web_ui/src/components/BaseConfigTree.tsx

```diff
@@ -43,6 +43,9 @@ function renderInput(inputComponents: Record<string, InputComponent>, def: lib.F
 		return <Checkbox/>;
 	}
 	if (def.type === "string") {
+		if (def.credential) {
+			return <Input.Password/>;
```

Comment:
With the addition of a password input, browsers will recognise this as a login form and attempt auto complete and ask the user if their want to save the password. This can be fixed using `<Input.Password autocomplete="new-password"/>` ([src](https://developer.mozilla.org/en-US/docs/Web/Security/Practical_implementation_guides/Turning_off_form_autocompletion#managing_autofill_for_login_fields))

Reply:
I set the token to `section-factorio new-password` and the username to `section-factorio username`. Hopefully that should be well behaved if more credentials are added. To make this possible I added an autoComplete entry in the field definition which sets the autocomplete property. This also makes it possible for plugins to fiddle with it should they face conflicts with it.

Reply:
My browser no longer auto fills on page load, when previously it did. As discussed on discord there doesn't seam to be a perfect solution and this is good enough.